include/mbedtls/config_psa.h

```diff
@@ -32,7 +32,13 @@
  * before we deduce what built-ins are required. */
 #include "psa/crypto_adjust_config_key_pair_types.h"
 
+#if defined(MBEDTLS_PSA_CRYPTO_CLIENT) && !defined(MBEDTLS_PSA_CRYPTO_C)
```

Comment:
I'm not sure I like the way this is written. How about:
```c
#if defined(MBEDTLS_PSA_CRYPTO_C)
/* If we are implementing PSA crypto ourselves, then we want to enable the required built-ins.
 * Otherwise, PSA features will be provided by the server. */
#include "mbedtls/config_adjust_legacy_from_psa.h"
#endif
```
The comment can be tuned, but my points are:
- I don't think we need to check for `CLIENT` here as I don't think we're ever reading `config_psa.h` with `CLIENT` disabled.
- Then I prefer "if condition then" to "if !condition else".